server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionUtil.java

```diff
@@ -551,16 +559,25 @@ private VariableWithLevelNode getVariableForConditional(VariableDefinitionNode d
     return new ResultWithErrors<>(variableNode, errors);
   }
 
-  private VariableWithLevelNode getPrecedingVariableForRedefine(
-      VariableDefinitionNode definitionNode, String redefineName) {
+  private static List<VariableWithLevelNode> getEligibleNodesForRedefine(
+      VariableDefinitionNode definitionNode) {
     Deque<Node> siblings = new LinkedList<>(definitionNode.getParent().getChildren());
-    while (!siblings.isEmpty()) {
-      Node precedingNode = siblings.pollLast();
-      if (precedingNode instanceof VariableWithLevelNode) {
-        VariableWithLevelNode variable = (VariableWithLevelNode) precedingNode;
-        if (!variable.isRedefines() || variable.getName().equals(redefineName)) return variable;
```

Comment:
on second look, it seems just adding a condition here for level_66 and level_88 would behave the same.